src/components/learner-credit-management/data/utils.js

```diff
@@ -653,25 +677,27 @@ export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime,
   const clonedCourseRuns = courseRuns.map(courseRun => ({
     ...courseRun,
     enrollBy: courseRun.hasEnrollBy ? dayjs.unix(courseRun.enrollBy).toISOString() : null,
+    enrollStart: courseRun.hasEnrollStart ? dayjs.unix(courseRun.enrollStart).toISOString() : null,
     upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
   }));
   const assignableCourseRunsFilter = ({
-    enrollBy, isActive, hasEnrollBy, isLateEnrollmentEligible,
+    enrollBy, enrollStart, start, isActive, hasEnrollBy, isLateEnrollmentEligible,
   }) => {
-    let isEligibleForEnrollment = true;
-    if (hasEnrollBy) {
-      const enrollByDate = dayjs(enrollBy);
-      // Determine eligibility based on the provided enrollBy is and the subsidy expiration date - refund threshold
-      isEligibleForEnrollment = (
-        !isDateBeforeToday(enrollByDate)
-        && enrollByDate.isBefore(minimumEnrollByDateFromToday({ subsidyExpirationDatetime }))
-      );
-      // Late redemption filter
-      if (isDateBeforeToday(enrollBy) && isLateRedemptionAllowed) {
-        const lateEnrollmentCutoff = dayjs().subtract(LATE_ENROLLMENTS_BUFFER_DAYS, 'days');
-        isEligibleForEnrollment = enrollByDate.isAfter(lateEnrollmentCutoff);
-        return isLateEnrollmentEligible && isEligibleForEnrollment;
-      }
+    // Determine eligibility based on the provided enrollBy, start, and enrollStart date
+    const isEligibleForEnrollment = isEnrollByDateWithinThreshold(
+      {
+        enrollBy,
+        isLateRedemptionAllowed,
+      },
+    ) && isStartDateWithinThreshold(
+      {
+        enrollStart,
+        start,
+        subsidyExpirationDatetime,
+      },
+    );
+    if (hasEnrollBy && isLateRedemptionAllowed && isDateBeforeToday(enrollBy)) {
+      return isLateEnrollmentEligible && isEligibleForEnrollment;
     }
     // General courseware filter
     return isActive && isEligibleForEnrollment;
```

Comment:
Nit: isEligibleForEnrollment is a common consideration, maybe return early if that's false? That way we can end the entire function with a sequence of checks and comments for ultimate readability.  Furthermore, you don't need to check hasEnrollBy or isDateBeforeToday(enrollBy) anymore because that's already rolled into isEligibleForEnrollment.
```suggestion
    if (!isEligibleForEnrollment) {
        // Basic checks against this content's critical dates and their relation to
        // the current date and subsidy expiration date have failed.
        return false;
    }
    if (isLateRedemptionAllowed) {
      // Special case: late enrollment has been enabled by ECS for this budget, and 
      // isEligibleForEnrollment already succeeded, so we know that late enrollment
      // would be happy given enrollment deadline of the course.  Now all we need
      // to do is make sure the run itself is generally eligible for late enrollment
      // (e.g. it is published and not archived, etc.)
      return isLateEnrollmentEligible;
    }
    // General courseware filter.  Content that's active is generally eligible for
    // enrollment (e.g. it is published and not archived and generally enrollable,
    // etc.).
    return isActive;
```

Reply:
Definitely leaving this open to put into consideration for a refined implementation. 👍🏽